packages/http/src/metadata.ts

```diff
@@ -195,6 +284,77 @@ export function getRequestVisibility(verb: HttpVerb): Visibility {
+ *
+ * When created with a verb, this provider will use the default visibility for that verb.
+ *
+ * @param verb - the HTTP verb for the operation
+ *
+ * @see {@link VisibilityProvider}
+ * @see {@link getParameterVisibilityFilter}
+ * @see {@link getReturnTypeVisibilityFilter}
+ */
+export function HttpVisibilityProvider(verb: HttpVerb): VisibilityProvider;
+/**
+ * A visibility provider for HTTP operations. Pass this value as a provider to the `getParameterVisibilityFilter` and
+ * `getReturnTypeVisibilityFilter` functions in the TypeSpec core to get the applicable parameter and return type
+ * visibility filters for an HTTP operation.
+ *
+ * When created with an options object, this provider will use the `verbSelector` function to determine the verb for the
+ * operation and use the default visibility for that verb, or the configured HTTP verb for the operation, and finally
+ * the GET verb if the verbSelector function is not defined and no HTTP verb is configured.
+ *
+ * @param options - an options object with a `verbSelector` function that returns the HTTP verb for the operation
+ *
+ * @see {@link VisibilityProvider}
+ * @see {@link getParameterVisibilityFilter}
+ * @see {@link getReturnTypeVisibilityFilter}
+ */
+export function HttpVisibilityProvider(options: OperationParameterOptions): VisibilityProvider;
+/**
+ * A visibility provider for HTTP operations. Pass this value as a provider to the `getParameterVisibilityFilter` and
+ * `getReturnTypeVisibilityFilter` functions in the TypeSpec core to get the applicable parameter and return type
+ * visibility filters for an HTTP operation.
+ *
+ * When created without any arguments, this provider will use the configured verb for the operation or the GET verb if
+ * no HTTP verb is configured and use the default visibility for that selected verb.
+ *
+ * @see {@link VisibilityProvider}
+ * @see {@link getParameterVisibilityFilter}
+ * @see {@link getReturnTypeVisibilityFilter}
+ */
+export function HttpVisibilityProvider(): VisibilityProvider;
+export function HttpVisibilityProvider(
+  verbOrParameterOptions?: HttpVerb | OperationParameterOptions,
+): VisibilityProvider {
+  const hasVerb = typeof verbOrParameterOptions === "string";
+
+  return {
+    parameters: (program, operation) => {
+      let verb = hasVerb
+        ? (verbOrParameterOptions as HttpVerb)
+        : (verbOrParameterOptions?.verbSelector?.(program, operation) ??
+          getOperationVerb(program, operation));
+
+      if (!verb) {
+        const [httpOperation] = getHttpOperation(program, operation);
+
+        verb = httpOperation.verb;
+      }
+
+      return visibilityToFilter(program, getDefaultVisibilityForVerb(verb));
+    },
+    returnType: (program, _) => {
+      const Read = getLifecycleVisibilityEnum(program).members.get("Read")!;
+      // For return types, we always use Read visibility in HTTP.
+      return { any: new Set([Read]) };
+    },
+  };
+}
+
 /**
```

Comment:
Do we want the default here to be to return a filter, so that this should be deprecated as well?

Reply:
Is this question about `HttpVisibilityProvider` or `resolveRequestVisibility`? If former, it will always return a filter. If latter, it would be severely breaking. I did consider it, but don't think we can really swallow deprecating the whole HTTP `Visibility` concept right now. I think instead of preemptively deprecating the HTTP-only `Visibility`, my hope is that if the core APIs are used in the next couple of months (i.e., people switch to using `getReturnTypeVisibilityFilter(..., HttpVisibilityProvider)` instead of `resolveRequestVisibility` or `MetadataInfo`, then HTTP's `Visibility` will become a vestigial feature that we can deprecate once some emitters have really proved that this analysis API works for their use case (I'm going to do this for the JS server emitter).